taggui/auto_captioning/models/joycaption.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import torch
+from transformers import LlavaForConditionalGeneration, BatchFeature
+
+from auto_captioning import captioning_thread
+from auto_captioning.auto_captioning_model import AutoCaptioningModel
+from utils.image import Image
+
+
+class JoycaptionLlavaLlama3(AutoCaptioningModel):
+    transformers_model_class = LlavaForConditionalGeneration
+
+    def __init__(self,
+                 captioning_thread_: 'captioning_thread.CaptioningThread',
+                 caption_settings: dict):
+        super().__init__(captioning_thread_, caption_settings)
+
+        self.input_length = None
+        self.dtype = torch.bfloat16
+        self.dtype_argument = ({'dtype': self.dtype}
+                               if self.device.type == 'cuda' else {})
+
+    def get_error_message(self) -> str | None:
+        if self.load_in_4_bit:
+            # The official model does not currently work when loaded
+            # in 4-bit mode.
+            return 'Joycaption cannot currently be loaded in 4-bit.'
+        return super().get_error_message()
+
+    @staticmethod
+    def get_default_prompt() -> str:
+        return 'Write a stable diffusion prompt for this image.'
+
+    def format_prompt(self, prompt: str) -> str:
+        conversation = [
+                {
+                    "role": "system", 
+                    "content": "You are a helpful image captioner.",
+                },
+                {
+                    "role": "user",
+                    "content": prompt.strip(),
+                },
+        ]
+        templated_prompt = self.processor.apply_chat_template(
+            conversation, tokenize=False, add_generation_prompt=True)
+
+        self.caption_start = self.caption_start.strip()
+        if self.caption_start:
+            templated_prompt = templated_prompt + self.caption_start
+
+        return templated_prompt
+
+    def get_additional_generation_parameters(self) -> dict:
+        additional_parameters = super().get_additional_generation_parameters()
+        additional_parameters['use_cache'] = True
+
+        tokenizer = self.get_tokenizer()
+        eos_token_id = (tokenizer('<|eot_id|>', add_special_tokens=False)
+                        .input_ids)[0]
+        additional_parameters['eos_token_id'] = eos_token_id
```

Comment:
Is there a reason you added this? I can't find it in the examples on Hugging Face and GitHub, and the model seems to work fine without it.